checkov/terraform/runner.py

```diff
@@ -116,47 +117,73 @@ def run(
         parsing_errors: dict[str, Exception] = {}
         self.load_external_checks(external_checks_dir)
         local_graph = None
-
+        all_graphs = []
         if self.context is None or self.definitions is None or self.breadcrumbs is None:
             self.definitions = {}
             logging.info("Scanning root folder and producing fresh tf_definitions and context")
             if root_folder:
                 root_folder = os.path.abspath(root_folder)
-
-                local_graph, self.definitions = self.graph_manager.build_graph_from_source_directory(
-                    source_dir=root_folder,
-                    local_graph_class=self.graph_class,
-                    download_external_modules=runner_filter.download_external_modules,
-                    external_modules_download_path=runner_filter.external_modules_download_path,
-                    parsing_errors=parsing_errors,
-                    excluded_paths=runner_filter.excluded_paths,
-                    vars_files=runner_filter.var_files,
-                    create_graph=CHECKOV_CREATE_GRAPH,
-                )
+                tf_split_graph = strtobool(os.getenv('TF_SPLIT_GRAPH', 'False'))
```

Comment:
as this part begins to be a bit big - what do you think about splitting the `if/else` block to a new function called `_generate_graphs_list` (or something like that)?

Reply:
I thought about it but the problem is here we are outputting the graph but also updating the self.definitions. So the definitions update will be like a silent side effect.